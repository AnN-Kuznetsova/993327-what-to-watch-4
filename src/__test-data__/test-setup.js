import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";

Enzyme.configure({
  adapter: new Adapter(),
});

const noop = () => {};
Object.defineProperty(window, `scrollTo`, {value: noop, writable: true});
Object.defineProperty(HTMLMediaElement.prototype, `play`, {value: noop, writable: true});
Object.defineProperty(HTMLMediaElement.prototype, `pause`, {value: noop, writable: true});
Object.defineProperty(HTMLMediaElement.prototype, `load`, {value: noop, writable: true});
