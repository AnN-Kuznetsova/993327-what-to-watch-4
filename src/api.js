import axios from "axios";


export const Error = {
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  VALIDATION: `VALIDATION`,
};


export const createAPI = (onFailRequest) => {
  const api = axios.create({
    baseURL: `https://4.react.pages.academy/wtw`,
    timeout: 1000 * 5,
    withCredentials: true,
  });

  const onSuccess = (response) => {
    return response;
  };

  const onFail = (error) => {
    onFailRequest(error);

    throw error;
  };

  api.interceptors.response.use(onSuccess, onFail);

  return api;
};
