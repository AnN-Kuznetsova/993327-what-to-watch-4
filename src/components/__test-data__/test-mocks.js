const promoMovie = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  releaseDate: new Date(2014, 0),
  posterUrl: `img/the-grand-budapest-hotel-poster.jpg`,
  backgroundUrl: `img/bg-the-grand-budapest-hotel.jpg`,
};

const movieTitles = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
  `Shutter Island`,
  `Pulp Fiction`,
  `No Country for Old Men`,
  `Snatch`,
  `Moonrise Kingdom`,
  `Seven Years in Tibet`,
  `Midnight Special`,
  `War of the Worlds`,
  `Dardjeeling Limited`,
  `Orlando`,
  `Mindhunter`,
  `Midnight Special`,
];

const genreName = `Kids & Family`;

const movieTitle = `Aviator`;


export {
  genreName,
  movieTitle,
  movieTitles,
  promoMovie,
};
