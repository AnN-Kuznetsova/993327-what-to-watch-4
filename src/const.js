const NUMBER_OF_CARDS_IN_CATALOG_AT_STARTUP = 8;
const NUMBER_OF_CARDS_TO_INCREMENT = 8;
const MAX_FILTER_ELEMENTS_COUNT = 10;
const NUMBER_OF_SIMILAR_FILMS = 4;
const VISIBLE_PARTICIPANTS_COUNT = 4;
const DELAY_PLAYBACK_PREVIEW = 1000;
const MIN_REVIEW_TEXT_LENGTH = 50;
const MAX_REVIEW_TEXT_LENGTH = 400;
const RATING_RANGE = 5;
const ERROR_COLOR = `#a8421e`;

const BASENAME_FOR_DEPLOY = `/htmlacademy-what-to-watch`;

const AppRoute = {
  MAIN: BASENAME_FOR_DEPLOY + `/`,
  SIGN_IN: BASENAME_FOR_DEPLOY + `/login`,
  MY_LIST: BASENAME_FOR_DEPLOY + `/myList`,
  FILM: BASENAME_FOR_DEPLOY + `/films/:id`,
  ADD_REVIEW: BASENAME_FOR_DEPLOY + `/films/:id/review`,
  PLAYER: BASENAME_FOR_DEPLOY + `/player/:id`,
};


export {
  DELAY_PLAYBACK_PREVIEW,
  ERROR_COLOR,
  MAX_FILTER_ELEMENTS_COUNT,
  MAX_REVIEW_TEXT_LENGTH,
  MIN_REVIEW_TEXT_LENGTH,
  NUMBER_OF_CARDS_IN_CATALOG_AT_STARTUP,
  NUMBER_OF_CARDS_TO_INCREMENT,
  NUMBER_OF_SIMILAR_FILMS,
  RATING_RANGE,
  VISIBLE_PARTICIPANTS_COUNT,
  AppRoute,
  BASENAME_FOR_DEPLOY,
};
