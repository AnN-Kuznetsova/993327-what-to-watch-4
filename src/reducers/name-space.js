export const NameSpace = {
  DATA: `DATA`,
  APPLICATION: `APPLICATION`,
  USER: `USER`,
};
