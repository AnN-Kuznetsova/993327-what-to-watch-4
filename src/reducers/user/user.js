import {extend} from "../../utils/utils";


const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  loginError: null,
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_LOGIN_ERROR: `SET_LOGIN_ERROR`,
};

const ActionCreator = {
  requireAuthorization: (status) => ({
    type: ActionType.REQUIRED_AUTHORIZATION,
    payload: status,
  }),
  setLoginError: (error) => ({
    type: ActionType.SET_LOGIN_ERROR,
    payload: error,
  }),
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then(() => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      })
      .catch((err) => {
        throw err;
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.email,
      password: authData.password,
    })
    .then(() => {
      dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      dispatch(ActionCreator.setLoginError(null));
    })
    .catch((error) => {
      dispatch(ActionCreator.setLoginError(error));
    });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload,
      });

    case ActionType.SET_LOGIN_ERROR:
      return extend(state, {
        loginError: action.payload
      });

    default:
      return state;
  }
};


export {
  ActionCreator,
  ActionType,
  AuthorizationStatus,
  Operation,
  reducer,
};
