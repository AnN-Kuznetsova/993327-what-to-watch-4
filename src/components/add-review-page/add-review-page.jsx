import PropTypes from "prop-types";
import React, {createRef, PureComponent} from "react";
import {connect} from "react-redux";

import {ActionCreator as ApplicationActionCreator} from "../../reducers/application/application";
import {ActionCreator as DataActionCreator, Operation} from "../../reducers/data/data";
import {Error} from "../../api";
import {Header} from "../header/header";
import {MIN_REVIEW_TEXT_LENGTH, MAX_REVIEW_TEXT_LENGTH, RATING_RANGE, PageType, AppRoute, ERROR_COLOR} from "../../const";
import {MoviePropType} from "../../prop-types";
import {RatingItem} from "../rating-item/rating-item";
import {Redirect} from "react-router-dom";
import {getDataError, getMovieById} from "../../reducers/data/selectors";
import {withNewReview} from "../../hocs/with-new-review/with-new-review";


const getErrorMessage = (dataError) => {
  switch (true) {
    case dataError === null:
      return null;

    case dataError.response && dataError.response.status === Error.UNAUTHORIZED:
      return (`Only authorized users can leave a review. Please register.`);

    case dataError.response === Error.VALIDATION:
      const errorTexts = [`Please enter correct data:`];
      if (dataError.data.ratingValueError) {
        errorTexts.push(`The rating of the film must be at least 1 star.`);
      }
      if (dataError.data.reviewTextValueError) {
        errorTexts.push(`Review text must be at least 50 and no more than 400 characters.`);
      }
      return errorTexts.join(`\n`);

    case dataError.response && dataError.response.status === Error.BAD_REQUEST:
    default:
      return (`Sorry, your review could not be submitted. Please try again.`);
  }
};


const getReviewRatingValidation = (ratingContainer) => {
  const ratingElement = Array.from(ratingContainer.querySelectorAll(`input`))
    .filter((ratingItem) => ratingItem.checked)[0];

  const rating = ratingElement ? +(ratingElement.value) : null;

  return rating ? rating : false;
};

const getReviewTextValidation = (reviewTextValue) => {
  return reviewTextValue.length >= MIN_REVIEW_TEXT_LENGTH && reviewTextValue.length <= MAX_REVIEW_TEXT_LENGTH ? reviewTextValue : false;
};


class AddReviewPageComponent extends PureComponent {
  constructor(props) {
    super(props);

    this._ratingRef = createRef();
    this._reviewTextRef = createRef();
    this._addReviewButtonRef = createRef();
    this._addReviewFormRef = createRef();

    this._ratingValue = null;
    this._reviewTextValue = null;

    this.handleAddRreviewPageOpen = this.handleAddRreviewPageOpen.bind(this);
    this.handleDataReviewChange = this.handleDataReviewChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount() {
    this.handleAddRreviewPageOpen();
  }

  componentDidUpdate() {
    this.handleAddRreviewPageOpen();
  }

  handleAddRreviewPageOpen() {
    if (this.props.movie) {
      this.props.openAddReviewPage(this.props.movie);
    }
  }

  handleDataReviewChange() {
    this._ratingValue = getReviewRatingValidation(this._ratingRef.current) || null;
    this._reviewTextValue = getReviewTextValidation(this._reviewTextRef.current.value) || null;

    this.props.onChange(this._ratingValue, this._reviewTextValue);

    if (this._ratingValue && this._reviewTextValue) {
      this._addReviewButtonRef.current.disabled = false;
      this._addReviewButtonRef.current.style.opacity = 1;
      this.props.setDataError(null);
    } else {
      this._addReviewButtonRef.current.disabled = true;
      this._addReviewButtonRef.current.style.opacity = 0.5;
      this.props.setDataError({
        response: Error.VALIDATION,
        data: {
          reviewTextValueError: this._reviewTextValue ? null : true,
          ratingValueError: this._ratingValue ? null : true,
        }
      });
    }
  }

  handleSubmit(event) {
    event.preventDefault();

    this.props.sendReview({
      movieId: this.props.movie.id,
      rating: this.props.reviewRating,
      comment: this.props.reviewText,
      addReviewFormElements: this._addReviewFormRef.current.elements,
    });
  }

  render() {
    const {
      movie,
      dataError,
      onError,
    } = this.props;

    const errorMessage = dataError ? getErrorMessage(dataError) : null;

    if (!movie) {
      onError();
      return (<Redirect to={AppRoute.MAIN} />);
    }

    return (
      <section className="movie-card movie-card--full"
        style={{backgroundColor: movie.backgroundColor}}>
        <div className="movie-card__header">
          <div className="movie-card__bg">
            <img src={movie.backgroundUrl} alt={movie.title} />
          </div>

          <Header />

          <div className="movie-card__poster movie-card__poster--small">
            <img src={movie.posterUrl} alt={`${movie.title} poster`} width="218" height="327" />
          </div>
        </div>

        <div className="add-review">
          <form
            action="#"
            className="add-review__form"
            onSubmit={this.handleSubmit}
            ref={this._addReviewFormRef}
          >
            <div className="rating">
              <div
                ref={this._ratingRef}
                className="rating__stars"
                style={dataError && dataError.response === Error.VALIDATION && dataError.data.ratingValueError ? {borderRadius: `8px`, boxShadow: `0 0 0 1px ${ERROR_COLOR}`} : {}}>
                {new Array(RATING_RANGE).fill(``).map((ratingItem, index) => (
                  <RatingItem
                    key={ratingItem + index}
                    id={index + 1}
                    onClick={this.handleDataReviewChange}
                  />
                ))}
              </div>
            </div>

            <div className="add-review__text" style={{backgroundColor: `rgba(255, 255, 255, 0.3)`}}>
              <textarea
                ref={this._reviewTextRef}
                className="add-review__textarea"
                name="review-text" id="review-text"
                placeholder="Review text"
                onChange={this.handleDataReviewChange}
                style={dataError && dataError.response === Error.VALIDATION && dataError.data.reviewTextValueError ? {borderRadius: `8px`, boxShadow: `0 0 0 1px ${ERROR_COLOR}`} : {}}>
              </textarea>
              <div className="add-review__submit">
                <button
                  ref={this._addReviewButtonRef}
                  className="add-review__btn"
                  type="submit"
                  disabled={true}
                  style={{opacity: 0.5}}
                >Post</button>
              </div>
            </div>
          </form>

          {errorMessage && <p style={{whiteSpace: `pre-wrap`, color: ERROR_COLOR}}>{errorMessage}</p>}
        </div>

      </section>
    );
  }
}


AddReviewPageComponent.propTypes = {
  routeProps: PropTypes.object.isRequired,
  movie: MoviePropType,
  dataError: PropTypes.object,
  sendReview: PropTypes.func.isRequired,
  setDataError: PropTypes.func.isRequired,
  openAddReviewPage: PropTypes.func.isRequired,
  reviewRating: PropTypes.number,
  reviewText: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onError: PropTypes.func.isRequired,
};


const mapStateToProps = (state, props) => {
  const {routeProps} = props;
  const movieId = +routeProps.match.params.id;

  return {
    movie: getMovieById(state, movieId),
    dataError: getDataError(state),
  };
};

const mapDispatchToProps = (dispatch) => ({
  setDataError(error) {
    dispatch(DataActionCreator.setDataError(error));
  },
  sendReview(reviewData) {
    dispatch(Operation.sendReview(reviewData));
  },
  openAddReviewPage(movie) {
    dispatch(ApplicationActionCreator.changeActiveMovie(movie));
    dispatch(ApplicationActionCreator.changeActivePage(PageType.ADD_REVIEW));
  },
  onError() {
    dispatch(DataActionCreator.setDataError({status: 404}));
    dispatch(ApplicationActionCreator.changeActivePage(PageType.ERROR));
  },
});

const AddReviewPage = connect(mapStateToProps, mapDispatchToProps)(AddReviewPageComponent);
const AddReviewPageWithNewReview = withNewReview(AddReviewPage);


export {
  AddReviewPageComponent,
  AddReviewPage,
  AddReviewPageWithNewReview,
};
