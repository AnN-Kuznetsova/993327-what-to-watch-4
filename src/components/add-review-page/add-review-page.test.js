import React from "react";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";

import {AddReviewPage} from "./add-review-page";
import {AuthorizationStatus} from "../../reducers/user/user";
import {Error} from "../../api";
import {NameSpace} from "../../reducers/name-space";
import {PageType} from "../../const";

import {mockPromoMovie} from "../../__test-data__/test-mocks";


const mockStore = configureStore([]);

const store = mockStore({
  [NameSpace.APPLICATION]: {
    activeMovie: mockPromoMovie,
    activePage: PageType.ADD_REVIEW,
  },
  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.AUTH,
  },
});

const nodeMock = {
  createNodeMock: () => {
    return {};
  }
};

const props = {
  movie: mockPromoMovie,
  dataError: null,
  sendReview: () => {},
  setDataError: () => {},
  reviewRating: null,
  reviewText: null,
  onChange: () => {},
};


describe(`Render AddReviewPage`, () => {
  it(`AddReviewPage should match with snapshot when dataError is null`, () => {
    const addReviewPageSnapshot = renderer.create(
        <Provider store={store} >
          <AddReviewPage {...props} />
        </Provider>, nodeMock
    ).toJSON();

    expect(addReviewPageSnapshot).toMatchSnapshot();
  });


  it(`AddReviewPage should match with snapshot when dataError is "BAD_REQUEST"`, () => {
    props.dataError = {
      response: {
        status: 400,
      }
    };

    const addReviewPageSnapshot = renderer.create(
        <Provider store={store} >
          <AddReviewPage {...props} />
        </Provider>, nodeMock
    ).toJSON();

    expect(addReviewPageSnapshot).toMatchSnapshot();
  });


  it(`AddReviewPage should match with snapshot when dataError is "UNAUTHORIZED"`, () => {
    props.dataError = {
      response: {
        status: 401,
      }
    };

    const addReviewPageSnapshot = renderer.create(
        <Provider store={store} >
          <AddReviewPage {...props} />
        </Provider>, nodeMock
    ).toJSON();

    expect(addReviewPageSnapshot).toMatchSnapshot();
  });


  it(`AddReviewPage should match with snapshot when dataError is "VALIDATION"`, () => {
    props.dataError = {
      response: Error.VALIDATION,
      data: {
        reviewTextValueError: true,
        ratingValueError: true,
      }
    };

    const addReviewPageSnapshot = renderer.create(
        <Provider store={store} >
          <AddReviewPage {...props} />
        </Provider>, nodeMock
    ).toJSON();

    expect(addReviewPageSnapshot).toMatchSnapshot();
  });
});
