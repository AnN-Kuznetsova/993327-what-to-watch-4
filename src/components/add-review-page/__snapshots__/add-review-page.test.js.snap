// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render AddReviewPage AddReviewPage should match with snapshot when dataError is "BAD_REQUEST" 1`] = `
<section
  className="movie-card movie-card--full"
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head false"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link false"
          href="/"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="#"
              onClick={[Function]}
            >
              The Grand Budapest Hotel
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="The Grand Budapest Hotel poster"
        height="327"
        src="img/the-grand-budapest-hotel-poster.jpg"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action="#"
      className="add-review__form"
      onSubmit={[Function]}
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
          style={Object {}}
        >
          <input
            className="rating__input"
            id="star-1"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={1}
          />
          <label
            className="rating__label"
            htmlFor="star-1"
          >
            Rating 
            1
          </label>
          <input
            className="rating__input"
            id="star-2"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={2}
          />
          <label
            className="rating__label"
            htmlFor="star-2"
          >
            Rating 
            2
          </label>
          <input
            className="rating__input"
            id="star-3"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={3}
          />
          <label
            className="rating__label"
            htmlFor="star-3"
          >
            Rating 
            3
          </label>
          <input
            className="rating__input"
            id="star-4"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={4}
          />
          <label
            className="rating__label"
            htmlFor="star-4"
          >
            Rating 
            4
          </label>
          <input
            className="rating__input"
            id="star-5"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={5}
          />
          <label
            className="rating__label"
            htmlFor="star-5"
          >
            Rating 
            5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.3)",
          }
        }
      >
        <textarea
          className="add-review__textarea"
          id="review-text"
          name="review-text"
          onChange={[Function]}
          placeholder="Review text"
          style={Object {}}
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            disabled={true}
            style={
              Object {
                "opacity": 0.5,
              }
            }
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
    <p
      style={
        Object {
          "whiteSpace": "pre-wrap",
        }
      }
    >
      Sorry, your review could not be submitted. Please try again.
    </p>
  </div>
</section>
`;

exports[`Render AddReviewPage AddReviewPage should match with snapshot when dataError is "UNAUTHORIZED" 1`] = `
<section
  className="movie-card movie-card--full"
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head false"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link false"
          href="/"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="#"
              onClick={[Function]}
            >
              The Grand Budapest Hotel
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="The Grand Budapest Hotel poster"
        height="327"
        src="img/the-grand-budapest-hotel-poster.jpg"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action="#"
      className="add-review__form"
      onSubmit={[Function]}
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
          style={Object {}}
        >
          <input
            className="rating__input"
            id="star-1"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={1}
          />
          <label
            className="rating__label"
            htmlFor="star-1"
          >
            Rating 
            1
          </label>
          <input
            className="rating__input"
            id="star-2"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={2}
          />
          <label
            className="rating__label"
            htmlFor="star-2"
          >
            Rating 
            2
          </label>
          <input
            className="rating__input"
            id="star-3"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={3}
          />
          <label
            className="rating__label"
            htmlFor="star-3"
          >
            Rating 
            3
          </label>
          <input
            className="rating__input"
            id="star-4"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={4}
          />
          <label
            className="rating__label"
            htmlFor="star-4"
          >
            Rating 
            4
          </label>
          <input
            className="rating__input"
            id="star-5"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={5}
          />
          <label
            className="rating__label"
            htmlFor="star-5"
          >
            Rating 
            5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.3)",
          }
        }
      >
        <textarea
          className="add-review__textarea"
          id="review-text"
          name="review-text"
          onChange={[Function]}
          placeholder="Review text"
          style={Object {}}
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            disabled={true}
            style={
              Object {
                "opacity": 0.5,
              }
            }
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
    <p
      style={
        Object {
          "whiteSpace": "pre-wrap",
        }
      }
    >
      Only authorized users can leave a review. Please register.
    </p>
  </div>
</section>
`;

exports[`Render AddReviewPage AddReviewPage should match with snapshot when dataError is "VALIDATION" 1`] = `
<section
  className="movie-card movie-card--full"
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head false"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link false"
          href="/"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="#"
              onClick={[Function]}
            >
              The Grand Budapest Hotel
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="The Grand Budapest Hotel poster"
        height="327"
        src="img/the-grand-budapest-hotel-poster.jpg"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action="#"
      className="add-review__form"
      onSubmit={[Function]}
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
          style={
            Object {
              "borderRadius": "8px",
              "boxShadow": "0 0 0 1px #A8421E",
            }
          }
        >
          <input
            className="rating__input"
            id="star-1"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={1}
          />
          <label
            className="rating__label"
            htmlFor="star-1"
          >
            Rating 
            1
          </label>
          <input
            className="rating__input"
            id="star-2"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={2}
          />
          <label
            className="rating__label"
            htmlFor="star-2"
          >
            Rating 
            2
          </label>
          <input
            className="rating__input"
            id="star-3"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={3}
          />
          <label
            className="rating__label"
            htmlFor="star-3"
          >
            Rating 
            3
          </label>
          <input
            className="rating__input"
            id="star-4"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={4}
          />
          <label
            className="rating__label"
            htmlFor="star-4"
          >
            Rating 
            4
          </label>
          <input
            className="rating__input"
            id="star-5"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={5}
          />
          <label
            className="rating__label"
            htmlFor="star-5"
          >
            Rating 
            5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.3)",
          }
        }
      >
        <textarea
          className="add-review__textarea"
          id="review-text"
          name="review-text"
          onChange={[Function]}
          placeholder="Review text"
          style={
            Object {
              "borderRadius": "8px",
              "boxShadow": "0 0 0 1px #A8421E",
            }
          }
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            disabled={true}
            style={
              Object {
                "opacity": 0.5,
              }
            }
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
    <p
      style={
        Object {
          "whiteSpace": "pre-wrap",
        }
      }
    >
      Please enter correct data:
The rating of the film must be at least 1 star.
Review text must be at least 50 and no more than 400 characters.
    </p>
  </div>
</section>
`;

exports[`Render AddReviewPage AddReviewPage should match with snapshot when dataError is null 1`] = `
<section
  className="movie-card movie-card--full"
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head false"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link false"
          href="/"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="#"
              onClick={[Function]}
            >
              The Grand Budapest Hotel
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="The Grand Budapest Hotel poster"
        height="327"
        src="img/the-grand-budapest-hotel-poster.jpg"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action="#"
      className="add-review__form"
      onSubmit={[Function]}
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
          style={Object {}}
        >
          <input
            className="rating__input"
            id="star-1"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={1}
          />
          <label
            className="rating__label"
            htmlFor="star-1"
          >
            Rating 
            1
          </label>
          <input
            className="rating__input"
            id="star-2"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={2}
          />
          <label
            className="rating__label"
            htmlFor="star-2"
          >
            Rating 
            2
          </label>
          <input
            className="rating__input"
            id="star-3"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={3}
          />
          <label
            className="rating__label"
            htmlFor="star-3"
          >
            Rating 
            3
          </label>
          <input
            className="rating__input"
            id="star-4"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={4}
          />
          <label
            className="rating__label"
            htmlFor="star-4"
          >
            Rating 
            4
          </label>
          <input
            className="rating__input"
            id="star-5"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={5}
          />
          <label
            className="rating__label"
            htmlFor="star-5"
          >
            Rating 
            5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.3)",
          }
        }
      >
        <textarea
          className="add-review__textarea"
          id="review-text"
          name="review-text"
          onChange={[Function]}
          placeholder="Review text"
          style={Object {}}
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            disabled={true}
            style={
              Object {
                "opacity": 0.5,
              }
            }
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;
